server/item/enchantment/vanishing.go

```diff
@@ -0,0 +1,55 @@
+package enchantment
+
+import (
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world"
+)
+
+// Vanishing is an enchantment that causes the item to disappear on death.
+type Vanishing struct{}
+
+// Name ...
+func (Vanishing) Name() string {
+	return "Vanishing"
```

Comment:
Would probably return `Curse of Vanishing` here to be consistent with Minecraft.

Reply:
Probably also a good idea to name the struct itself `CurseOfVanishing`.